src/VisualStudio/Core/SolutionExplorerShim/HierarchyItemToProjectIdMap.cs

```diff
@@ -47,6 +46,16 @@ public bool TryGetProjectId(IVsHierarchyItem hierarchyItem, string targetFramewo
             var project = _workspace.DeferredState.ProjectTracker.ImmutableProjects
                     .Where(p =>
                     {
+                        // We're about to access various properties of the IVsHierarchy associated with the project.
```

Comment:
who ends up using this API?  i.e. what features need this, and how do we know they're only for C# or VB?